main.py

```diff
@@ -55,105 +81,166 @@ def purchase_steps(browser):
     # wait until its visible and then click the purchase button
     logger.debug('find and click on the last purchase button')
     WebDriverWait(browser, TIMEOUT).until(
-        EC.visibility_of_element_located((By.XPATH, "//button[contains(@class,'btn-primary')]"))
+        EC.visibility_of_element_located((
+            By.XPATH,
+            "//button[contains(@class,'btn-primary')]"
+        ))
     ).click()
 
     # wait until its visible and then click the 'I Agree" popup
     # 'Refund and Right of Withdrawal Information' popup
     try:
         logger.debug('accept the conditions of refund popup')
         WebDriverWait(browser, TIMEOUT).until(
-            EC.visibility_of_all_elements_located(
-                (By.XPATH, "//button[contains(@class,'btn-primary')]"))
+            EC.visibility_of_all_elements_located((
+                By.XPATH,
+                "//button[contains(@class,'btn-primary')]"
+            ))
         )[1].click()
-    except (NoSuchElementException, TimeoutException, LookupError) as ex:
+
+    except (NoSuchElementException, TimeoutException, LookupError):
         logger.debug('no refund conditions popup to accept')
 
     logger.debug('Wait for confirmation that checkout is complete')
 
     # Purchase should be complete. Checking for confirmation
-    WebDriverWait(browser, TIMEOUT).until(EC.visibility_of_element_located((
-        By.XPATH,
-        "//h1/span[contains(text(),'Install')]|//span[contains(text(),'Thank you for buying')]"
-    )))
+    WebDriverWait(browser, TIMEOUT).until(
+        EC.visibility_of_element_located((
+            By.XPATH,
+            "//h1/span[contains(text(),'Install')]|"
+            "//span[contains(text(),'Thank you for buying')]"
+        ))
+    )
 
 
-def execute():
-    browser = open_browser()
-    browser.get(FREE_GAMES_PAGE_URL)
+def login(browser):
+    logger.debug('find and click on login button')
+    el = WebDriverWait(browser, TIMEOUT).until(
+        EC.element_to_be_clickable((
+            By.XPATH,
+            "//div[@id='user']/ul/li/a/span"
+        ))
+    )
+    el.click()
+
+    logger.debug('find and click on EpicGame login method')
+    el = WebDriverWait(browser, TIMEOUT).until(
+        EC.element_to_be_clickable((
+            By.XPATH,
+            "//div[@id='login-with-epic']/div[2]/span/h6"
+        ))
+    )
```

Comment:
On both of these lookups `EC.element_to_be_clickable()` if you choose to search the elements by their IDs why not go for the option `By.ID()`
I.G. `EC.element_to_be_clickable((By.ID, "user"))` and `EC.element_to_be_clickable((By.ID, "login-with-epic"))`

Reply:
Bruh, I always prefer xpath lmao

Reply:
Alright, not a problem with that 😄 
I still have to ask though, if you use the ID like that does it need the following tags?
`//div[@id='user']/ul/li/a/span` -> `//div[@id='user']`

Reply:
I don't know, in theory yes. To redo the xpath links I used [selenium IDE](https://chrome.google.com/webstore/detail/selenium-ide/mooikfkahbdckldjjndioackbalphokd), a browser extension.  it pretty much reads everything I click on the webpage and gives me the xpath I need